desktop/src/navigator.rs

```diff
@@ -160,10 +161,20 @@ impl NavigatorBackend for ExternalNavigatorBackend {
                 let client =
                     client.ok_or_else(|| Error::FetchError("Network unavailable".to_string()))?;
 
-                let isahc_request = match request.method() {
+                let mut isahc_request = match request.method() {
                     NavigationMethod::Get => IsahcRequest::get(processed_url.to_string()),
                     NavigationMethod::Post => IsahcRequest::post(processed_url.to_string()),
                 };
+                if let Some(headers) = isahc_request.headers_mut() {
+                    for (name, val) in request.headers().iter() {
+                        headers.insert(
+                            HeaderName::from_bytes(name.as_bytes())
```

Comment:
I see `HeaderName` implements [`FromStr`](https://docs.rs/http/latest/http/header/struct.HeaderName.html#impl-FromStr-for-HeaderName). I think it can be used for a bit cleaner code.